lib/job-iteration/iteration.rb

```diff
@@ -132,6 +133,10 @@ def interruptible_perform(*arguments)
 
       run_callbacks(:shutdown)
 
+      if @needs_reenqueue && !completed.nil?
```

Comment:
Just thinking about this, I think the `completed.nil?` check is pointless.  There can't be a `throw :abort` between when we set `@needs_reenqueue` and the return that's on the next line

Reply:
Agreed. `completed` should be `false` if `@ needs_reenqueue` is true.